src/core/qgscoordinatereferencesystem.cpp

```diff
@@ -1398,8 +1398,7 @@ bool QgsCoordinateReferenceSystem::readXml( const QDomNode &node )
         //make sure the map units have been set
         setMapUnits();
 
-        //@TODO this srs needs to be validated!!!
-        d->mIsValid = true; //shamelessly hard coded for now
+        d->mIsValid = OSRSetFromUserInput( d->mCRS, d->mAuthId.toLower().toLatin1() ) == OGRERR_NONE;
```

Comment:
This can be dropped entirely - including the current TODO. It's already handled a few lines above by the call to `setProj4String( myNode.toElement().text() );`